File: server/mahaprinting_server/DomainObjects/printer.py
class Printer:
    id: int
    url: str
    name: str
    apiKey: str

    def __init__(self, id: int, name: str, url: str, apiKey: str) -> None:
        self.id = id
        self.url = url
        self.name = name
        self.apiKey = apiKey

File: client/mahaprinting-client/src/ServerAPI/Printer.ts
import { observable } from "mobx";

export default class Printer {
  @observable id: number;
  @observable name: string;
  @observable url: string;
  @observable apiKey: string;
  @observable state: string;
  @observable jobInfo: { [key: string]: any };
  @observable model: string;

  constructor({
    id,
    name,
    url,
    apiKey,
    state,
    jobInfo,
    model,
  }: {
    id: number;
    name: string;
    url: string;
    apiKey: string;
    state: string;
    jobInfo: { [key: string]: any };
    model: string;
  }) {
    this.id = id;
    this.name = name;
    this.url = url;
    this.apiKey = apiKey;
    this.state = state;
    this.jobInfo = jobInfo;
    this.model = model;
  }
}
